res/text.f.glsl

```diff
@@ -31,8 +32,23 @@ void main()
         alphaMask = vec4(1.0);
         color = vec4(bg.rgb, 1.0);
     } else {
-        vec3 textColor = texture(mask, TexCoords).rgb;
-        alphaMask = vec4(textColor, textColor.r);
-        color = vec4(fg, 1.0);
+        if (colored != 0) {
+            // Color glyphs, like emojis.
+            vec4 glyphColor = texture(mask, TexCoords);
+            alphaMask = vec4(glyphColor.a);
+
+            // Handle premultiplied alpha.
+            //
+            // NOTE - we won't show garbage from division by zero on a screen,
+            // due to alpha being 0, so it's perfrectly fine to do so.
+            glyphColor.rgb = vec3(glyphColor.rgb / glyphColor.a);
```

Comment:
```suggestion
            // Revert alpha premultiplication.
            if (glyphColor.a != 0) {
                glyphColor.rgb = vec3(glyphColor.rgb / glyphColor.a);
            }
```

Reply:
Don't you like fun stuff?

Reply:
No fun allowed in Rust unfortunately. Especially because so many platforms out there.